nginx/kitspace.template

```diff
@@ -45,6 +45,33 @@ server {
 
 	server_name gitea.@KITSPACE_DOMAIN;
 
+
+	location = /api/v1/repos/migrate {
+        # this path is handled differently to increase the timeout for response
+        # as migration for repos may take longer than the default timeout
+
+		proxy_read_timeout 180s;
+
+		add_header                 'Access-Control-Allow-Origin'      '@KITSPACE_URL' always;
+		add_header                 'Access-Control-Allow-Methods'     'POST';
+		add_header                 'Access-Control-Allow-Credentials' 'true' always;
+		add_header                 'Access-Control-Allow-Headers'     'cache-control, pragma, user-agent';
+		proxy_cookie_path          ~^(.+)$                            '$1; domain=.@KITSPACE_DOMAIN';
+		proxy_set_header           Origin                             gitea.@KITSPACE_DOMAIN;
+		proxy_pass_request_headers on;
+
+		if ($request_method = OPTIONS) {
+			add_header Content-Length 0;
+			add_header Content-Type text/plain;
+			add_header 'Access-Control-Allow-Origin' '@KITSPACE_URL';
+			add_header 'Access-Control-Allow-Methods' 'POST';
+			add_header 'Access-Control-Allow-Headers' 'cache-control, pragma, user-agent, content-type';
+			add_header 'Access-Control-Allow-Credentials' 'true' always;
+			return 200;
+		}
+		proxy_pass http://gitea:3000;
+	}
+
```

Comment:
Why not just increas the timeout for all gitea requests?

Reply:
If any other request took that long to resolve this means there's something wrong. The official docs advise not to use more than 75s, the default is 60s. Though for `/api/v2/repos/migrate` we know it's normal to take more than that.